### DISCUSSION TITLE
Why include VSCode instead of VSCodium?

### DISCUSSION THREAD

**Discussion by Overwatching** [winpython/winpython#923] [General]
Is there a good reason to include the mainline VSCode with all the telemetry and other shenanigans instead of VSCodium?

For those that don't know what the difference between the two is:
As stated [here](https://github.com/Microsoft/vscode/issues/60#issuecomment-161792005)

The VSCode code base is open source. It is MIT licensed.
- Microsoft
  - takes the source and builds it with a configuration script that adds their custom telemetry endpoints.
  - releases the build. It is no longer MIT licensed.
- VSCodium project
  - takes the source and builds it with a configuration script that **DOES NOT** add telemetry endpoints.
  - releases the build. It is **STILL** MIT licensed.

That's all it is. A pre-compiled binary.
https://vscodium.com/#why

Was this a conscious decision?
Or was this an oversite?

**Comment by stonebig** [comment]
Because trust and support is more important than free as in beer. Otherwise users would be on Linux.

**Reply by stonebig** [reply]
I didn't check if Debian has vscodium inside. Would be interesting to check if trust is an option

**Reply by Overwatching** [reply]
> Because trust and support is more important than free as in beer. Otherwise users would be on Linux.

1. Both are free as in beer. But, while the VSCode project SOURCE is MIT license, the pre-compiled Microsoft Binaries are NOT because Microsoft adds in their telemetry. However, VSCodium is just the base VSCode source, and maintains the MIT license.
2. Why is trust an issue? All VSCodium does is compile the VSCode source without adding the extra telemetry that Microsoft adds to their own releases. All they are doing is providing binaries so people don't have to compile it themselves. Not everyone who might want an IDE can/should be compiling a whole project from source.
If anything, VSCode is less trustworthy since there is plenty of telemetry and other stuff in it that is only vaguely included in any EULA.
3. Support? It is the same code...A normal user literally couldn't tell the difference between the two if you covered up the name. The only major difference is that it uses it's own extension repo. But every extension I have looked at is ALSO uploaded to the VSCodium one. It's exactly the same one-click install. And on the off chance an extension is NOT on their repo, there is an obvious button on the MS repo to just download it, and a button in VSCodium to manually install from file (which is also in VSCode so it isn't even custom code).

We aren't talking about VSCode vs VSCodium for your great grandmother.
We are talking about a user who is going to be programming something, and knows enough to go to a web site.

**Comment by stonebig** [comment]
Same for computers, people will purchase an apple, a Dell, an HP for the brand, even if chongkingwhatever does also white label pc of same, or better, quality

**Reply by stonebig** [reply]
Brand is the most valuable asset of any company/project, and is earned after many years of proven reliability.

**Reply by stonebig** [reply]
Can be lost in a day, also.

**Reply by stonebig** [reply]
Brand has not the same perceived value if your a true pc specialist, or student with little money. But otherwise, the value of a brand has a meaningfull reason.

**Reply by Overwatching** [reply]
Brand?

If a user doesn't ALREADY know the difference between VSCode and VSCodium, there is almost no chance they even know they aren't the same thing.

If a user actually DOES know the difference between the two, they might actually care about telemetry.

Let's take an example here since most people don't really seem to understand what the difference between the two is.

Imagine 2 video doorbells. The 'Bring' and the 'BringIt'.
Both doorbells are made from the same injection molded plastic. From the same physical molds.
Both have the same circuit board inside. Identical.
Both are priced exactly the same.
Both connect to software that is exactly the same, except for the name.

But.

The Bring doorbell comes hardcoded to send a photo back to the manufacturer every time the doorbell is used. And the manufacturer includes an extra part in their EULA stating that they can use the pictures however they want, including selling them to anyone who will pay for them. 

The BringIt doorbell does not.

Which doorbell do you get?

**Reply by Overwatching** [reply]
I realize this is a few months old by now (It's been a busy semester) but I wanted to follow up on this to clarify a few things.

I don't think I properly described the difference between VSCode and VSCodium.

As stated [here](https://github.com/Microsoft/vscode/issues/60#issuecomment-161792005)

The VSCode code base is open source. It is MIT licensed.
- Microsoft
  - takes the source and builds it with a configuration script that adds their custom telemetry endpoints.
  - releases the build. It is no longer MIT licensed.
- VSCodium project
  - takes the source and builds it with a configuration script that **DOES NOT** add telemetry endpoints.
  - releases the build. It is **STILL** MIT licensed.

That's all it is. A pre-compiled binary.
https://vscodium.com/#why

**Comment by RoyiAvital** [comment]
What about having the Telemetry disabled by default for VSCode in WinPython?
That would basically give the same result, right?

**Reply by Overwatching** [reply]
I'm not sure what you mean.

WinPython has VSCode bundled/integrated. But the WinPython project doesn't change VSCode.

The VSCodium project **IS** the VSCode project, but without the telemetry being added.

Were you suggesting that WinPython just change the settings of VSCode when it get's installed?
Because most of the telemetry in VSCode cannot be disabled. That's why the VSCodium project exists. To remove it.
